libphal/phal_dump.C

```diff
@@ -169,473 +219,574 @@ void writeSbeData(const std::filesystem::path& dumpPath, uint32_t reasonCode,
  */
-void checkSbeState(struct pdbg_target* pib)
+void checkSbeState(struct pdbg_target* pib_fsi, const int sbeTypeId)
 {
 	enum sbe_state state;
-	if (sbe_get_state(pib, &state)) {
+
+	// If the SBE dump is already collected return error
+	auto rv = -1;
+	if (PROC_SBE_DUMP == sbeTypeId)
+		rv = sbe_get_state(pib_fsi, &state);
+	else if (ODYSSEY_SBE_DUMP == sbeTypeId)
+		rv = sbe_ody_get_state(pib_fsi, &state);
+
+	if (rv) {
 		log(level::ERROR, "Failed to read SBE state information (%s)",
-		    pdbg_target_path(pib));
+		    pdbg_target_path(pib_fsi));
 		throw sbeError_t(exception::SBE_STATE_READ_FAIL);
 	}
+
 	if (state == SBE_STATE_FAILED) {
 		log(level::ERROR,
-		    "Dump is already collected from the SBE on (%s)",
-		    pdbg_target_path(pib));
+		    "Dump is already collected from the "
+		    "SBE on (%s)",
+		    pdbg_target_path(pib_fsi));
 		throw sbeError_t(exception::SBE_DUMP_IS_ALREADY_COLLECTED);
 	}
 }
 
 /**
- * @brief Executes the SBE extract RC and writes recovery action data.
- * @param[in] proc The processor target on which to execute the SBE extract RC.
- * @param[in] dumpPath The filesystem path where the SBE data and recovery
- * action information will be written.
+ * @brief Sets the SBE state
  *
- * @throw std::runtime_error Throws if the SBE extract RC operation fails.
+ * @param pib_fsi The pib or fsi target depending upon the chip
+ * @param sbeTypeId The chip type ID
  */
-void executeSbeExtractRc(struct pdbg_target* proc,
-			 const std::filesystem::path& dumpPath)
+int setSbeState(struct pdbg_target* pib_fsi, const int sbeTypeId,
+		const sbe_state& state)
 {
-	P10_EXTRACT_SBE_RC::RETURN_ACTION recovAction;
-	fapi2::ReturnCode fapiRc = p10_extract_sbe_rc(proc, recovAction, true);
-	log(level::INFO,
-	    "p10_extract_sbe_rc for proc=%s returned rc=0x%08X and SBE "
-	    "Recovery Action=0x%08X",
-	    pdbg_target_path(proc), fapiRc, recovAction);
-	writeSbeData(dumpPath, fapiRc, recovAction);
+	// Set SBE state to state mode
+	if (PROC_SBE_DUMP == sbeTypeId)
+		return sbe_set_state(pib_fsi, state);
+	else if (ODYSSEY_SBE_DUMP == sbeTypeId)
+		return sbe_ody_set_state(pib_fsi, state);
+	else
+		return -1;
 }
 
 /**
- * @brief Writes SBE register values to a file.
- * @param[in] dumpRegs A vector of `DumpSBERegVal` structures containing the SBE
- * register values to be written.
- * @param[in] basePath The filesystem path of the file to which the data is to
- * be written.
+ * @brief Extracts SBE RC from the corrupted chip and writes the info into a
+ * dump file
  *
- * @throw std::runtime_error Throws if there is an error opening the file or
- * during the file write operation.
+ * @param target The chip target
+ * @param dumpPath The path of the dump file
+ * @param sbeTypeId The chip type, i.e.; proc or OCMB
  */
-void writeSBERegValuesToFile(const std::vector<DumpSBERegVal>& dumpRegs,
-			     const std::filesystem::path& basePath)
+void extractSbeRc(struct pdbg_target* target,
```

Comment:
why this function name changed?

Reply:
you may change the name as appropriate but not in this commit

Reply:
This is a different function than `writeSBERegValuesToFile`. I didn't get you here.

Reply:
extractSbeRc, why this changed?

Reply:
It was `executeExtractSbeRc` I guess. I removed "execute" because it sounds redundant (as any function call is made to execute the logic within it). Open to discuss it further.